### ISSUE TITLE
Can't import module after install [Anaconda]

### ISSUE THREAD

**Issue by EWouters** [uoip/pangolin#10]
Hi, I compiled as per the instructions in the readme, (had to install `libeigen3-dev` to get it to compile) and installed after fixing the typo in the setup.py script. However I still can't import pangolin to python. Can anyone help me with this issue?

Thanks!

```
(base) user@pc:~/GitHub/twitchslam/build/pangolin$ python setup.py install
running install
copying ./pangolin.cpython-36m-x86_64-linux-gnu.so -> /home/user/anaconda3/lib/python3.5/site-packages
(base) user@pc:~/GitHub/twitchslam/build/pangolin$ python
Python 3.5.4 |Anaconda custom (64-bit)| (default, Nov 20 2017, 18:44:38) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pangolin
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named 'pangolin'
```

**Comment by plasticuproject** [comment]
The module isn't there. If you go to lib/python/site-packages/ I suspect you will not find that module in there.

So your setup/install failed, maybe because of a missing dependency. Read the documentation, make sure you install all the dependencies and run setup install again. There has to be a dependency missing somewhere.

Edit:
 I just noticed that it installed it to your '/home/user/anaconda3/lib/python3.5/site-packages' directory. 
If you're using a virtual environment, maybe it is coping the module to the wrong directory?

**Comment by EWouters** [comment]
Hi, thanks for your quick response!

Turns out I 'ignored' the warning below. I did not actually ignored it, I ran `python -m pip install pybind11` to install the dependency, but this did not remove the warning so I proceeded anyway. Also it says in the readme that pybind11 is not required to be installed.

Just now I found [this](https://github.com/pybind/pybind11/issues/698#issuecomment-427429162) issue where someone said to install it using a specific pip. That didn't help, but as I'm using Anaconda I ran `conda install pybind11`, after which the warning disappeared.

```
user@pc:~/GitHub/twitchslam/build/pangolin/build$ cmake ..
Build type not set (defaults to release)
-DCMAKE_BUILD_TYPE=Debug for debug
-- Python Found and Enabled
-- Eigen Found and Enabled
-- V4L Found and Enabled
-- ffmpeg Found and Enabled
-- libpng Found and Enabled
-- libjpeg Found and Enabled
-- libtiff Found and Enabled
-- libopenexr Found and Enabled
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 

CMake Warning at python/CMakeLists.txt:3 (find_package):
  By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "pybind11",
  but CMake did not find one.

  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring done
-- Generating done
```

**Comment by EWouters** [comment]
So I rechecked all the dependencies and I think I have all of them installed.

Also the library seems to be installed in the right place. But still does not import.

```
user@pc:~/GitHub/twitchslam/build/pangolin$ ls /home/user/anaconda3/lib/python3.5/site-packages | grep pangolin
pangolin.cpython-36m-x86_64-linux-gnu.so
user@pc:~/GitHub/twitchslam/build/pangolin$ python -c "import sys; print(sys.path)"
['', '/home/user/anaconda3/lib/python35.zip', '/home/user/anaconda3/lib/python3.5', '/home/user/anaconda3/lib/python3.5/plat-linux', '/home/user/anaconda3/lib/python3.5/lib-dynload', '/home/user/anaconda3/lib/python3.5/site-packages']
user@pc:~/GitHub/twitchslam/build/pangolin$ python -c "import pangolin"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named 'pangolin'

```

**Comment by plasticuproject** [comment]
Hmm, if the module is in the right path you shouldn't get that import error. Have you tried installing another module to that environment and checked to see if it imports correctly?

**Comment by EWouters** [comment]
Yes, I am able to use other projects that have more complete installers.

I seem to have located the problem. The library that is build is for python 3.6, which is my system python, and then copied to my (python 3.5) Anaconda environment.

Maybe I need to pass the location of my python binary to the CMake script...

**Comment by plasticuproject** [comment]
If that's the case, see the Closed issue #11, that may help you. And then you should probably close this issue as well.

**Comment by EWouters** [comment]
Thanks @plasticuproject!
